fields/types/categorySet/CategorySet.js

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import Select from 'react-select';
+import xhr from 'xhr';
+import async from 'async';
+
+import Field from '../Field';
+
+const categorySetStyle = {
+  display: 'flex',
+  flexDirection: 'row',
+  marginBottom: '20px',
+};
+
+const categoryMenuStyle = {
+  flexGrow: 1,
+  marginRight: '16px',
+};
+
+const subcategoryMenuStyle = {
+  flexGrow: 1
+};
+
+const btnContainerStyle = {
+  display: 'flex',
+  flexDirection: 'row',
+  justifyContent: 'flex-end'
+};
+
+const btnStyle = {
+  border: 'none',
+  backgroundColor: 'transparent'
+};
+
+module.exports = Field.create({
+
+  displayName: 'RelationshipField',
+
+  getInitialState() {
+    let value = [];
+    if (Array.isArray(this.props.value)) {
+      this.props.value.forEach(categorySet => {
+        if (categorySet && categorySet.category) {
+          value.push({ category: categorySet.category, subcategory: categorySet.subcategory });
+        }
+      });
+    }
+    return {
+      categoryOptions: null,
+      subcategoryOptionsMap: null,
+      value: value.length === 0 ? [{ category: undefined, subcategory: undefined }] : value,
+    };
+  },
+
+  componentDidMount() {
+    const getSubcategoryOptions = (subcategoryIDs, callback) => {
+      async.map(subcategoryIDs, (subcategoryID, callback) => {
+        xhr({
+          url: Keystone.adminPath + `/api/tags?basic&search=${subcategoryID}`,
+          responseType: 'json',
+        }, (err, resp, data) => {
+          if (err || !data || !Array.isArray(data.results) || data.results.length <= 0) {
+            console.error('Error loading items:', err);
+            return callback(err);
+          }
+          callback(null, { value: data.results[0].id, label: data.results[0].name });
+        });
+      }, callback);
+    };
+
+    xhr({
+      url: Keystone.adminPath + '/api/post-categories?basic',
+      responseType: 'json',
+    }, (err, resp, data) => {
+      if (err || !data || !Array.isArray(data.results)) {
+        console.error('Error loading items:', err);
+        return;
+      }
+      let categoryOptions = [];
+      let subcategoryOptionsMap = new Map([]);
+      async.each(data.results, (category, callback) => {
+        if (category && category.id && category.name && category.fields && Array.isArray(category.fields.subcategory) && category.fields.subcategory.length > 0) {
+          categoryOptions.push({ value: category.id, label: category.name });
+          const subcategoryIDs = category.fields.subcategory;
+          getSubcategoryOptions(subcategoryIDs, (err, results) => {
+            if (err) {
```

Comment:
Question: Would we consider to retry `getSubcategoryOptions` function or hint user to reload if `getSubcategoryOptions` fail ?
On the side of user, category set cannot be done without subcategory options.

Reply:
alert added in https://github.com/twreporter/keystone/pull/220/commits/4f6a9726546c3f5517856ec16f5370280950f2d8